app/models/labware_creators/pooled_wells_by_sample.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require_dependency 'labware_creators/base'
+
+module LabwareCreators
+  # This labware creator pools PBMC isolations with the same samples in pairs
+  # (or configured number of source wells) per destination before cell counting
+  # to reduce the number runs on Celleca (cell counting). The robot transfer is
+  # done from LRC Blood Bank to LRC PBMC Bank plate. Only the wells with passed
+  # state will be transferred to the destination plate. The destination wells
+  # are compressed to top left by column on the plate.
+  class PooledWellsBySample < Base
+    include SupportParent::PlateOnly
+
+    # By default, source wells with the sample are pooled in pairs.
+    DEFAULT_NUMBER_OF_SOURCE_WELLS = 2
+
+    # Number of source wells with the same sample to be pooled.
+    def number_of_source_wells
+      purpose_config[:number_of_source_wells] || DEFAULT_NUMBER_OF_SOURCE_WELLS
+    end
+
+    def well_filter
+      @well_filter ||= WellFilter.new(creator: self)
+    end
+
+    def filters=(filter_parameters)
+      well_filter.assign_attributes(filter_parameters)
+    end
+
+    # Parent plate using SS v2 API
+    def source_plate
+      @source_plate ||= Sequencescape::Api::V2::Plate.find_by(uuid: parent.uuid)
+    end
+
+    # List of passed wells from parent plate
+    def passed_parent_wells
+      source_plate.wells.select { |well| well.state == 'passed' }
+    end
+
+    # List of passed wells from parent plate in column order
+    def parent_wells_in_columns
+      passed_parent_wells.sort_by(&:coordinate)
+    end
+
+    # List of pools to be created; index is the destination pool number; each pool is a list of source wells
+    def build_pools
+      parent_wells_in_columns
+        .group_by { |well| well.aliquots.first.sample.uuid }
+        .flat_map { |_uuid, wells| wells.each_slice(number_of_source_wells).to_a }
+    end
+
+    # List of pools built from passed wells from parent plate
+    def pools
+      @pools ||= build_pools
+    end
+
+    # Object mapping source wells to destination wells for transfers
+    def transfer_hash
+      result = {}
+      pools.each_with_index do |pool, index|
+        dest_location = WellHelpers.well_at_column_index(index)
+        pool.each do |source_well|
+          source_location = source_well.location
+          result[source_location] = { dest_locn: dest_location }
+        end
+      end
+      result
+    end
+
+    # Well object at well location on plate
+    def get_well_for_plate_location(plate, well_location)
+      plate.wells.detect { |well| well.location == well_location }
+    end
+
+    # Attributes for transfer request from source_well to dest_plate
+    def request_hash(source_well, dest_plate)
+      dest_location = transfer_hash[source_well.location][:dest_locn]
+      {
+        'source_asset' => source_well.uuid,
+        'target_asset' => get_well_for_plate_location(dest_plate, dest_location)&.uuid,
+        # 'submission' => submission,
```

Comment:
The `submission_id` comes in through the `well_filter.filtered` method as `[well, additional_params]` where additional_params is `{.'submission_id': 1 }`, it extracts the submission id from the wells. It is normal to have it in the transfers.